### PR TITLE
Bump importlib metadata version, part 2

### PR DESCRIPTION
**Pull request by maresb**
This is a continuation of #153 since @samj1912's status is "On Vacation till 25th April" and I'd like to see this fixed sooner. (This is a blocker for using cruft in Conda environments since Conda is more strict about dependency requirements.)

@samj1912 already did the work, @rxm7706 suggested the bugfix, so let's see if the tests pass now.

(Note that I'm continuing from @samj1912's verified commits, so I'm properly extending from his branch.)

### CODE REVIEW

**Comment by maresb** [comment]
Hmm, it looks like @samj1912 is currently the only active maintainer, so I assume we'll have to wait for him to return before a merge can take place.

Also, I wonder if it was intentional to remove Python 3.6 from the test matrix while the pin in `tool.poetry.dependencies` is `^3.6`. For consistency, it seems like either the pin should be updated to `^3.7` or we should be testing 3.6.

**Comment by maresb** [comment]
Ping @adamcunnington-mlg

**Comment by adamcunnington-mlg** [comment]
I think we have to wait for his return, sorry.

**Comment by samj1912** [comment]
We would also need to fix the import path at https://github.com/cruft/cruft/blob/0c6de85d974197969c0e65913857eaa36b788e5e/cruft/_version.py#L1 to use the standard library importlib when available and fallback to the backport when not.

File: cruft/_commands/utils/generate.py
Comment (samj1912) [inline cruft/_commands/utils/generate.py:39]: When would this

Comment (maresb) [reply cruft/_commands/utils/generate.py:39]: Never, it is to appease mypy

Comment (samj1912) [reply cruft/_commands/utils/generate.py:39]: Can we do an assert then? That conveys the intention better (asserting the internal state is consistent) rather than checking and raising a runtime error. That should also fix mypy issues.

Comment (samj1912) [inline cruft/_commands/utils/generate.py:39]:
```suggestion
    assert repo.working_dir is not None
```

Comment (maresb) [reply cruft/_commands/utils/generate.py:39]: That's what I did originally, but then bandit raised a warning that assert statements may be ignored during runtime.

Comment (maresb) [reply cruft/_commands/utils/generate.py:39]:
```
+ poetry run bandit -r cruft/
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.8.12
Run started:2022-04-20 16:51:04.089010

Test results:
>> Issue: [B101:assert_used] Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
   Severity: Low   Confidence: High
   Location: cruft/_commands/utils/generate.py:39:4
   More Info: https://bandit.readthedocs.io/en/latest/plugins/b101_assert_used.html
38
39          assert repo.working_dir is not None
40          context = _generate_output(cruft_state, Path(repo.working_dir), cookiecutter_input, output_dir)
```

@samj1912, do you still prefer the assert approach?

Comment (samj1912) [reply cruft/_commands/utils/generate.py:39]: Happy to add either a bandit or mypy ignore statement then :) https://bandit.readthedocs.io/en/latest/config.html#exclusions

Comment (maresb) [reply cruft/_commands/utils/generate.py:39]: Thanks! This is my first time seeing bandit.

**Review by samj1912 — APPROVED** [review]

**Comment by samj1912** [comment]
Thank you so much for this PR! Sorry for any CI relayed issues you had to solve as well.

**Comment by maresb** [comment]
Yaay!!! Thanks so much @samj1912!!!!

**Comment by samj1912** [comment]
@maresb looks like the portray version we are using for docs is incompatible with the mkdocs release which is causing our docs workflow to fail. This is blocking the release.Would you be willing to submit a PR to fix this?

**Comment by samj1912** [comment]
Logs available at https://github.com/cruft/cruft/runs/6100661586?check_suite_focus=true

**Comment by maresb** [comment]
@samj1912, ya, I just noticed that as well... I'll see if I can find a quick fix...

**Pull request merged**
